### DISCUSSION TITLE
INPUT IN THE FORM OF STRING

### DISCUSSION THREAD

**Discussion by DIVANSHMAHAJAN** [speechly/speechly#136] [Q&A]
I want to take input from the user in the form of string .
 
text="";
*send_message {Add} text $(text) 

I have added text to entities and send_message to intents. $SPEECHLY.STRING(text) also gives error.

**Comment by thisismatu** [comment]
Hey @DIVANSHMAHAJAN 👋 

If you have a know set of words you'd like to get from the user, you could write it like this:

```
texts = [apples | oranges]
*send_message {Add} $texts(text)
```

In the example above i have an Intent `send_message` and an Entity `text` of type `string`.

For a more comprehensive documentation on the SAL syntax, please refer to our docs https://docs.speechly.com/slu-examples/cheat-sheet/

**Reply by DIVANSHMAHAJAN** [reply]
But i want to gather everything the user says after Add .. 
I am using it for chat application

**Reply by aukkonen** [reply]
Hello @DIVANSHMAHAJAN 
The general principle for what you want is the same as in the above answer. Instead of a list of example words, you would define a list of example phrases. like this:
 ```
texts = [
  hi | bye | yeah | nope | see you later | take care | I'm on my way | tonight is good | are you free this saturday
  that's great | how are you | do you need anything from the grocery store | remind me to feed the cat
]
*send_message {add} $texts(text)
```
Note that the phrases specified are just examples. The system should, once deployed, support also other things. However, the more of these examples you give, the better will the system work. (The list above is probably a bit too short, you might want to add a few dozen more.)

However, you might not even need to use an entity here. Alternatively you could just use the entire transcript as the message. Kind of like is done in this example where the transcript is used as a search phrase: https://codepen.io/speechly/pen/VwzoMrW?editors=1010

**Reply by DIVANSHMAHAJAN** [reply]
https://codepen.io/speechly/pen/VwzoMrW?editors=1010
In this also while speaking if we give a pause then previous spoken words get erased off
CODE:
{segment ?(
<>
{segment.words.map((w)=>w.value).join(" ")}
</>
) : null}

HOW TO GET AROUND THIS? till the time the button is pressed I want to hear the full msg..
whereas https://demos.speechly.com/speech-to-text/ in this it doesnt happen so..

**Comment by DIVANSHMAHAJAN** [comment]
https://codepen.io/speechly/pen/VwzoMrW?editors=1010 
In this also while speaking if we give a pause then previous spoken words get erased off 
CODE:
{segment ?(
      <>
      {segment.words.map((w)=>w.value).join(" ")}
      </>
      ) : null}

HOW TO GET AROUND THIS? till the time the button is pressed I want to hear the full msg..
whereas https://demos.speechly.com/speech-to-text/ in this it doesnt happen so..

**Reply by aukkonen** [reply]
> https://codepen.io/speechly/pen/VwzoMrW?editors=1010 In this also while speaking if we give a pause then previous spoken words get erased off CODE: {segment ?( <> {segment.words.map((w)=>w.value).join(" ")} </> ) : null}
> 
> HOW TO GET AROUND THIS? till the time the button is pressed I want to hear the full msg.. whereas https://demos.speechly.com/speech-to-text/ in this it doesnt happen so..

Please see https://github.com/speechly/speechly/discussions/141
